From: Jan Kara <jack@suse.cz>
Subject: readahead-add-trace-points-v2
Date: Tue, 9 Sep 2025 16:58:50 +0200

Move tracepoint from do_page_cache_ra() to page_cache_ra_unbounded() as
that is a more standard function.

Link: https://lkml.kernel.org/r/20250909145849.5090-2-jack@suse.cz
Signed-off-by: Jan Kara <jack@suse.cz>
Cc: Pankaj Raghav <kernel@pankajraghav.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/trace/events/readahead.h |    2 +-
 mm/readahead.c                   |    3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

--- a/include/trace/events/readahead.h~readahead-add-trace-points-v2
+++ a/include/trace/events/readahead.h
@@ -11,7 +11,7 @@
 #include <linux/fs.h>
 #include <linux/pagemap.h>
 
-TRACE_EVENT(do_page_cache_ra,
+TRACE_EVENT(page_cache_ra_unbounded,
 	TP_PROTO(struct inode *inode, pgoff_t index, unsigned long nr_to_read,
 		 unsigned long lookahead_size),
 
--- a/mm/readahead.c~readahead-add-trace-points-v2
+++ a/mm/readahead.c
@@ -228,6 +228,8 @@ void page_cache_ra_unbounded(struct read
 	 */
 	unsigned int nofs = memalloc_nofs_save();
 
+	trace_page_cache_ra_unbounded(mapping->host, index, nr_to_read,
+				      lookahead_size);
 	filemap_invalidate_lock_shared(mapping);
 	index = mapping_align_index(mapping, index);
 
@@ -317,7 +319,6 @@ static void do_page_cache_ra(struct read
 	loff_t isize = i_size_read(inode);
 	pgoff_t end_index;	/* The last page we want to read */
 
-	trace_do_page_cache_ra(inode, index, nr_to_read, lookahead_size);
 	if (isize == 0)
 		return;
 
_
